internal/datacoord/compaction_trigger.go

```diff
@@ -594,6 +594,12 @@ func (t *compactionTrigger) handleGlobalSignal(signal *compactionSignal) error {
 }
 
 func (t *compactionTrigger) handleClusteringCompactionSignal(signal *compactionSignal) error {
+	ts, err := t.allocTs()
+	if err != nil {
+		log.Warn("allocate ts failed, skip to handle compaction")
+		return err
+	}
+
 	t.forceMu.Lock()
```

Comment:
can this lock be removed ?

Reply:
not recommend to remove it. Only one handleXXXCompactionSignal can do at the same time. The following logic are all memory updates.